core/src/main/scala/kafka/log/LogSegment.scala

```diff
@@ -95,6 +97,9 @@ class LogSegment private[log] (val log: FileRecords,
   /* the number of bytes since we last added an entry in the offset index */
   private var bytesSinceLastIndexEntry = 0
 
+  /* whether or not we need to flush the parent dir during flush */
```

Comment:
during flush => during the first flush ?

Reply:
Not really. We changed the value of `atomicNeedsFlushParentDir` after the first flush. The value of `needsFlushParentDir` in the construction function is for during the first flush. Do you have any suggestions on how to comment on them?

Reply:
Changed it to `during the next flush`.